apps/github-api-integration/behaviours/validate-name.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const Model = require('../models/github');
+
+module.exports = config => {
+
+  return superclass => class extends superclass {
+
+    // The process method is part of the POST pipeline. For details
+    // see https://github.com/UKHomeOfficeForms/hof-behaviour-hooks.
+    process(req, res, next) {
+
+      const name = req.form.values.name;
+
+      // Define an instance of the Model
+      // with attributes and config (uri)
+      const model = new Model({name}, config);
+
+      model.fetch()
+        .then(result => {
+          // Save the user name
+          // and continue
+          req.sessionModel.set('name', name);
+          super.process(req, res, next);
+        })
+        .catch(err => {
+          if (err.status === 404) {
+            // if the user doesn't exist
+            // render a 400 error template
+            // to inform the user that the
+            // request was invalid
+            return res.status(400).render('400', {
+              name: name,
+              baseUrl: req.baseUrl
+            });
```

Comment:
Anyway, I'd possibly go for a validation error rather than a whole page render here.